docs/specifications/broker-authentication.md

```diff
@@ -0,0 +1,107 @@
+# Broker Authentication
+
+This document attempts to outline the high-level plan for adding authentication
+to the broker api routes.
+
+When talking to service brokers, the [Service Catalog uses basic auth](#service-catalog-basic-auth). The Ansible Service Broker will implement Basic Auth using a secret shared between the catalog and the broker. The authentication will be configurable allowing a user i.e. developer to disable it.
+
+The plan for this sprint is to do the following:
+
+* make auth routes configurable by enhancing the configuration
+```
+auth:
+   - type: basic | INSERT OTHER TYPE HERE
+   - enable: true | false
```

Comment:
I don't really see a reason to restrict the broker section after N fields (or any section for that matter). What I *would* want to see is nice comments for each explaining what they do inside the example config. Personally, I think this makes sense in the broker section.

The config file is mainly for broker operators, they probably don't even know what a handler is. `broker.auth` makes sense to that audience though.